src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -652,8 +671,19 @@ private void AddGenericTypeParametersAndConstraintsCustomAttributes(EntityHandle
             }
         }
 
-        private void AddDefaultValue(EntityHandle parentHandle, object? defaultValue) =>
+        private void AddDefaultValue(EntityHandle parentHandle, object? defaultValue)
+        {
+            if (defaultValue != null)
+            {
+                Type type = defaultValue.GetType();
+                if (type.IsEnum)
+                {
+                    defaultValue = Convert.ChangeType(defaultValue, type.GetEnumUnderlyingType());
```

Comment:
Was this added for convenience of the caller or is it necessary (change Enum to to the underlying type)?

Reply:
It was throwing otherwise, and it was suggested in ECMA spec, I'll add a comment here